### ISSUE TITLE
DefaultViewModelFactory is not used by "default" as of 3.1.3

### ISSUE THREAD

**Issue by ZackOPP** [InsertKoinIO/koin#1277]
**Is your feature request related to a problem? Please describe.**
I noticed that the `ComponentCallbacks.getViewModel()` is using the following default `owner` parameter:
```
owner = { ViewModelOwner.from(this as ViewModelStoreOwner, this as? SavedStateRegistryOwner) }
```
Because it's passing a value for the `ViewModelOwner#stateRegistry` property, the `Scope.pickFactory` method returns a `ViewModelProdiver.Factory` of type `StateViewModelFactory`. I'd expect that the **_default_** implementation of `ComponentCallbacks.getViewModel()` uses the `DefaultViewModelFactory` instead. Otherwise, why is it called `Default**`?

**Describe the solution you'd like**
In order to keep the same default behavior as 3.1.2 and earlier versions, could you change the default parameter as follows?
```
owner = { ViewModelOwner.from(this as ViewModelStoreOwner) }
```

I'm having different **crashes** when testing fragments so far with `androidx.fragment.app.testing.launchFragment`. I tried to pass the `owner` parameter as mentioned above and the crashes are gone. Rather than updating all my `getViewModel` calls, I'd like to check if there's any plan to set the `DefaultViewModelFactory` as the **_default_** one again.

**Update**
The crashes that I'm getting are the same as this issue (https://github.com/InsertKoinIO/koin/issues/936). It seems it's an issue with how ViewModels are mocked rather than the factory used by default.

**Comment by ZackOPP** [comment]
**Update**
The crashes that I'm getting are the same as this issue (https://github.com/InsertKoinIO/koin/issues/936). It seems it's an issue with how ViewModels are mocked rather than the factory used by default.

**Comment by arnaudgiuliani** [comment]
Fixed in 3.2.0 - https://github.com/InsertKoinIO/koin/blob/3.2.x/android/koin-android/src/main/java/org/koin/androidx/viewmodel/ViewModelResolver.kt
